app/src/main/java/org/oppia/android/app/devoptions/devoptionsitemviewmodel/DeveloperOptionsModifyLessonProgressViewModel.kt

```diff
@@ -1,7 +1,31 @@
 package org.oppia.android.app.devoptions.devoptionsitemviewmodel
 
+import org.oppia.android.app.devoptions.RouteToMarkChaptersCompletedListener
+import org.oppia.android.app.devoptions.RouteToMarkStoriesCompletedListener
+import org.oppia.android.app.devoptions.RouteToMarkTopicsCompletedListener
+
 /**
  * [DeveloperOptionsItemViewModel] to provide features to modify lesson progress such as
  * marking chapters completed, marking stories completed and marking topics completed.
  */
-class DeveloperOptionsModifyLessonProgressViewModel : DeveloperOptionsItemViewModel()
+class DeveloperOptionsModifyLessonProgressViewModel(
+  private val routeToMarkChaptersCompletedListener: RouteToMarkChaptersCompletedListener,
+  private val routeToMarkStoriesCompletedListener: RouteToMarkStoriesCompletedListener,
+  private val routeToMarkTopicsCompletedListener: RouteToMarkTopicsCompletedListener
+) : DeveloperOptionsItemViewModel() {
+
+  /** Called when the 'mark chapters completed' button is clicked by the user. */
```

Comment:
I think `called when button is clicked` can be replaced by `routes the user to the <screen_name> screen for <use-case>. ` Ditto for others as well.

Reply:
KDoc updated. PTAL.